### PR TITLE
EnumerationField

### PR DESCRIPTION
**Pull request by agriffard**
Allows to add an EnumerationField specifying options and choosing the kind of editor like a select dropdown, a radiobuttons list or a checkboxes list.

### CODE REVIEW

**Comment by agriffard** [comment]
Selecting multiple checkboxes doesn't work for the moment.

I don't know how to bind the selected checkboxes to the Value of the EnumerationField.

If a SelectedValues has to be used, is it in the Field or in the EditEnumerationViewModel?

The ListBox Editor is also missing, for the same reason.

Settings can also be improved: 
AllowMultiple: Is it a setting or does it depend of the editor?
Required
DefaultValue
Options with Text;Value

Do I have to make an editor with a jquery plugin like Bootstrap-select or Select2?

**Comment by sebastienros** [comment]
What if enumeration was actually an editor on text field?

**Comment by matiasmolleja** [comment]
I'm thinking that maybe including the enumerator into the text field can cause problems in the admin?
I mean, if an user is looking for an enumerator field when defining a new content type, he probably won't find it easily if it is hidden as one of the text field editors.

**Comment by sebastienros** [comment]
The physical type (how it's stored) should be either a single value of a list value. This is true for each type of value  ({string, numeric, bool, ...). In this case if you want multiple strings then we need a "MultipleText" field, and some editors to manage how to edit these: drop down, fancy javascript, comma separated, ...

The advantage is that you can change the editor afterwards without loosing any data. And also we can let the UI be extensible per field.

File: src/OrchardCore.Modules/Orchard.ContentFields/ViewModels/EditEnumerationFieldViewModel.cs [hunk 1]
```diff
@@ -7,6 +7,25 @@ namespace Orchard.ContentFields.ViewModels
     public class EditEnumerationFieldViewModel
     {
         public string Value { get; set; }
+        public string[] SelectedValues
+        {
+            get
+            {
+                if (Field != null)
+                {
+                    return Field.SelectedValues;
```
Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.ContentFields/ViewModels/EditEnumerationFieldViewModel.cs]: This is a bad pattern. Copy the values in the driver.

**Comment by sebastienros** [comment]
Beta 2

Can you remove the custom serialization in the field with ';', we just need a string[] as is.

**Comment by agriffard** [comment]
Closed because outdated

**Pull request closed without merging**